### ISSUE TITLE
`::exists()` method is `unsafe` ?

### ISSUE THREAD

**Issue by poliorcetics** [thisKai/rust-windows-dll#6]
I don't understand why this method is unsafe and there is no documentation for it on https://docs.rs.

**Comment by roblabla** [comment]
I agree, right now it's very confusing, as there's no indication of which invariants we must uphold when calling `exists`. From skimming through the code, I believe that there is no such invariant, and exists is always safe to call (and as such should be declared safe).

**Comment by thisKai** [comment]
The exists method is only unsafe because it might have to load the library itself, which calls the win32 api function LoadLibraryExW. I don't know if there can be any guarantee that a call to this function is safe, so I decided to err on the side of caution.
The function doesn't use the return value from LoadLibraryExW so if that counts as safe then I could remove the unsafe keyword in the next version.

**Comment by roblabla** [comment]
I think it's safe to assume that, so long as `LoadLibraryExW` itself is being used correctly (e.g. with a proper null-terminated string), then it is sound.

Calling LoadLibraryExW may, as a side-effect, run arbitrary code (such as the DllMain) which could cause soundness issues. But IMO, this lives in the same realm as "Writing to /dev/mem can be used to SIGSEGV a safe rust program", and is, in practice, unlikely to lead to problems.

As such, I do think that it would be OK to remove the `unsafe` keyword there.
